doc/Process_Console.md

```diff
@@ -0,0 +1,589 @@
+ Process Console
+ ===============
+ 
+ `process_console` is a capsule that implements a small shell over the UART that allows
+ a terminal to inspect the kernel and control userspace processes.
+
+Commands
+ --------
+
+ This module provides a simple text-based console to inspect and control
+ which processes are running. The console has eleven commands:
+  - `help` - prints the available commands and arguments
+  - `status` - prints the current system status
+  - `list` - lists the current processes with their IDs and running state
+  - `stop n` - stops the process with name n
+  - `start n` - starts the stopped process with name n
+  - `fault n` - forces the process with name n into a fault state
+  - `terminate n` - terminates the running process with name n, moving it to the Terminated state
+  - `boot n` - tries to restart a Terminated process with name n
+  - `panic` - causes the kernel to run the panic handler
+  - `kernel` - prints the kernel memory map
+  - `process n` - prints the memory map of process with name n
+
+ ### `list` Command Fields:
+
+ - `PID`: The identifier for the process. This can change if the process
+   restarts.
+ - `Name`: The process name.
+ - `Quanta`: How many times this process has exceeded its allotted time
+   quanta.
+ - `Syscalls`: The number of system calls the process has made to the kernel.
+ - `Restarts`: How many times this process has crashed and been restarted by
+   the kernel.
+ - `Grants`: The number of grants that have been initialized for the process
+   out of the total number of grants defined by the kernel.
+ - `State`: The state the process is in.
+
+ Setup
+ -----
+
+ You need a device that provides the `hil::uart::UART` trait. This code
+ connects a `ProcessConsole` directly up to USART0:
+
+ ```rust
+ # use kernel::{capabilities, hil, static_init};
+ # use capsules::process_console::ProcessConsole;
+
+ pub struct Capability;
+ unsafe impl capabilities::ProcessManagementCapability for Capability {}
+
+ let pconsole = static_init!(
+     ProcessConsole<usart::USART>,
+     ProcessConsole::new(&usart::USART0,
+                  115200,
+                  &mut console::WRITE_BUF,
+                  &mut console::READ_BUF,
+                  &mut console::COMMAND_BUF,
+                  kernel,
+                  Capability));
+ hil::uart::UART::set_client(&usart::USART0, pconsole);
+
+ pconsole.start();
+ ```
```

Comment:
Having a setup section is ok, but it needs to be updated with an example of using the component. I doubt this code would even work anymore.

Reply:
You can take a look at how microbit instantiates it.

Reply:
I updated the setup, but I don't know if I did it correctly.